### PR TITLE
Change api to user hapi.js

### LINKED ISSUE #4: Replace Express with Hapi for API

### PR DESCRIPTION
**Pull request by garrensmith**
Converts api to use hapi.js instead of Express.
I've changed how `start-server.js` works, it now calls a function and gets the http
listener from that. This allows it to be a little more generic and it
works better with express. Also remove error handling route as hapi has
that all built in.

This fixes #4

### CODE REVIEW

File: api/index.js [hunk 1]
```diff
@@ -1,27 +1,47 @@
...
+export function createServer({ port }) {
+  server.connection({
+    port,
+    host: '0.0.0.0',
+    router: {
+      stripTrailingSlash: true,
+    },
+  });
 
+  server.register([
```
Comment (BenJanecke) [inline api/index.js:16]: Could we perhaps pull this out into a file/module of sorts?

File: api/index.js [hunk 2]
```diff
@@ -1,27 +1,47 @@
...
-});
-app.use(bodyParser.json());
-app.use(bodyParser.urlencoded({ extended: false }));
-app.use(express.static(path.join(__dirname, 'public')));
-app.use(routes);
-app.use(errors);
+export function createServer({ port }) {
+  server.connection({
+    port,
+    host: '0.0.0.0',
```
Comment (BenJanecke) [inline api/index.js:10]: How does this work with deploys, does this assume like an nginx server?

File: api/routes/index.js [hunk 3]
```diff
@@ -1,4 +1,27 @@
-import express from 'express';
-const router = express.Router();
+export function register(server, options, next) {
+  /* Serve static files from public folder */
+  server.route({
```
Comment (BenJanecke) [inline api/routes/index.js]: This should probably also be in it's own module, we are going for endpoint style design.

Comment (BenJanecke) [inline api/routes/index.js]: By that I mean the public stuff, the index route should stay here and the file should then in turn import child routes/endpoints.

File: api/index.js (hunk 2 above)
Comment (garrensmith) [inline api/index.js:10]: That should work for anything. I have tested that config on heroku and with docker.

File: api/routes/index.js (hunk 3 above)
Comment (garrensmith) [inline api/routes/index.js]: I'm happy to move the public endpoint out. Ideally I was thinking the register function in `api/index` would pull in each route file.

Comment (garrensmith) [inline api/routes/index.js]: Kind of like how we have done it here. https://github.com/RedCometLabs/hapi-react-couch/blob/master/server.js#L44

File: api/index.js (hunk 2 above)
Comment (BenJanecke) [inline api/index.js:10]: :+1:

File: api/routes/index.js (hunk 3 above)
Comment (BenJanecke) [inline api/routes/index.js]: Perfect

File: api/index.js (hunk 1 above)
Comment (garrensmith) [inline api/index.js:16]: Pulling this into another file/module doesn't feel right yet. I would like to see how this works out first. Unless you have other ideas.

Comment (BenJanecke) [inline api/index.js:16]: Thats fine then

**Comment by BenJanecke** [comment]
:+1: merge when ready

**Pull request merged**